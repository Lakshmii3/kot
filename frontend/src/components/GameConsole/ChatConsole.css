.chat_console {
  display:flex;
  
  flex-direction: column;
  font-family: 'Open Sans', sans-serif;
 
  color: black;
  }

  .submit_game_command{
    color: black;
  }