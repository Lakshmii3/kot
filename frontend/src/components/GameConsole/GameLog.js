import React from 'react';

import { List } from 'antd';
import "./GameLog.css";

const GameLog = (props) => {
    return (
        <div>
            <List
                // header={<div>Header</div>}
                // footer={<div>Footer</div>}
                bordered
                dataSource={props.data}
                renderItem={item => (
                    <List.Item>
                        {item}
                    </List.Item>
                )}
            />
        </div>
    );
}


export default GameLog;