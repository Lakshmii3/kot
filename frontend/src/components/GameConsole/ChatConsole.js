import React from 'react';
import "./ChatConsole.css";
import Button from "react-bootstrap/Button";

// import GameLog from './GameLog'

import ChatInstance from './../../services/chatService'


const chatLogExample = [
  '-- Sammy turn â€“',
  'Rolled: E E H 1 2 3',
  'Decision: keep E E',
  'Rolled: E P 1 2 2',
  'Decision: keep E',
  'Rolled: 2 2 2 (TRIPPLE! +2 VP)',
  ':: Gain: E E E +2 VP ::',
  'Buy: Psychic Probe',
  'End Turn'
];

class ChatConsole extends React.Component {

  constructor(props) {
    super(props);

    ChatInstance.connect();

    this.waitForSocketConnection(() => {
      ChatInstance.initChatUser(this.props.currentUser);
      ChatInstance.addCallbacks(this.setMessages.bind(this), this.addMessage.bind(this))
      ChatInstance.fetchMessages(this.props.currentUser);
    });

    this.state = {
      username: props.username,
      gameRoom: props.gameRoom,
      loggedIn: true,
      log: chatLogExample, // this holds the name of each list
      cmd: "enter cmd"
    };

    // this.handleChange = this.handleChange.bind(this);
    // this.SubmitGameCommand = this.SubmitGameCommand.bind(this);
  }

  // SubmitGameCommand(e) {
  //   let data = this.state.log;
  //   data.push(this.state.cmd);

  //   // TO DO : Call dice REST endpoint requesting reroll.
  //   //this.props.sendMessage(this.state.message)

  //   this.setState({
  //     log: data
  //   })
  // }

  // handleChange(e) {
  //   this.setState({
  //     cmd: e.target.value
  //   })
  // }


  waitForSocketConnection(callback) {
    const component = this;
    setTimeout(
      function () {
        // Check if websocket state is OPEN
        if (ChatInstance.state() === 1) {
          console.log("Connection is made")
          callback();
          return;
        } else {
          console.log("wait for connection...")
          component.waitForSocketConnection(callback);
        }
      }, 100); // wait 100 milisecond for the connection...
  }

  componentDidMount() {
    this.scrollToBottom();
  }

  componentDidUpdate() {
    this.scrollToBottom();
  }

  scrollToBottom = () => {
    const chat = this.messagesEnd;
    const scrollHeight = chat.scrollHeight;
    const height = chat.clientHeight;
    const maxScrollTop = scrollHeight - height;
    chat.scrollTop = maxScrollTop > 0 ? maxScrollTop : 0;
  }

  addMessage(message) {
    this.setState({ messages: [...this.state.messages, message] });
  }

  setMessages(messages) {
    this.setState({ messages: messages.reverse() });
  }

  messageChangeHandler = (event) => {
    this.setState({
      message: event.target.value
    })
  }

  sendMessageHandler = (e, message) => {
    const messageObject = {
      from: this.props.currentUser,
      text: message
    };
    ChatInstance.newChatMessage(messageObject);
    this.setState({
      message: ''
    })
    e.preventDefault();
  }

  renderMessages = (messages) => {
    const currentUser = this.props.currentUser;
    return messages.map((message, i) => <li key={message.id} className={message.author === currentUser ? 'me' : 'him'}> <h4 className='author'>{message.author} </h4><p>{message.content}</p></li>);
  }

  render() {
    const messages = this.state.messages;
    const currentUser = this.props.currentUser;
    return (
      <div className='chat'>
        <div className='container'>
          <h1>Chatting as {currentUser} </h1>
          <h3>Displaying only the last 50 messages</h3>
          <ul ref={(el) => { this.messagesEnd = el; }}>
            {
              messages &&
              this.renderMessages(messages)
            }
          </ul>
        </div>
        <div className='container message-form'>
          <form onSubmit={(e) => this.sendMessageHandler(e, this.state.message)} className='form'>
            <input
              type='text'
              onChange={this.messageChangeHandler}
              value={this.state.message}
              placeholder='Type a Message'
              required />
            <Button variant="btn btn-secondary" className='submit' type='submit' value='Submit'>
              Send
            </Button>
          </form>
        </div>
      </div>
    );
  }

  // render() {
  //   return (
  //     <div className="chat_console">
  //       <GameLog data={this.state.log} />

  //       <form className="submit_game_command"
  //         onSubmit={this.SubmitGameCommand}
  //         onChange={this.handleChange}>
  //         <textarea value={this.state.value} onChange={this.handleChange}>Type Command [enter]</textarea>
  //       </form>
  //       <button onClick={this.SubmitGameCommand}>
  //         Submit Comand
  //         </button>
  //     </div>
  //   )
  // }
}

export default ChatConsole;