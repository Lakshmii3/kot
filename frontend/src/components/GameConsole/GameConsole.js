import React from 'react';

import { List, Typography } from 'antd';
import "./GameConsole.css";

import GameLog from './GameLog'

var rollExample = [
  'heart',
  'energy',
  'paw',
  '1',
  '2',
  '3'
];

class GameConsole extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      log: rollExample, // this holds the name of each list
      cmd: "enter cmd"
    };

    this.OnChange = this.OnChange.bind(this);
    this.SubmitGameCommand = this.SubmitGameCommand.bind(this);
  }

  SubmitGameCommand(e) {
    let data = this.state.log;
    data.push(this.state.cmd);

    // TO DO : Call dice REST endpoint requesting reroll.
    //this.props.sendMessage(this.state.message)

    this.setState({
      log: data
    })
  }

  OnChange(e) {
    this.setState({
      cmd: e.target.value
    })
  }

  render() {
    return (
      <div className="game_console">
        <GameLog data={this.state.log} />
        <form
          onSubmit={this.SubmitGameCommand}
          onChange={this.OnChange}
          className="submit_game_command"></form>
        <button onClick={this.SubmitGameCommand}>
          Submit Comand
          </button>
      </div>
    )
  }
}

export default GameConsole;