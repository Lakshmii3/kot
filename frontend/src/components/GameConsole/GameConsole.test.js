import React from 'react';
import { shallow } from 'enzyme';

import GameConsole from './GameConsole';

describe('Verify GameConsole layout template', () => {
    it('GameConsole should render correctly', () => {
        const errorPage = shallow(<GameConsole />);
        expect(errorPage).toMatchSnapshot();
    });
})