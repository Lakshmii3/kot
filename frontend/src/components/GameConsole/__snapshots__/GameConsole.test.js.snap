// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify GameConsole layout template GameConsole should render correctly 1`] = `
<div
  className="game_console"
>
  <GameLog
    data={
      Array [
        "heart",
        "energy",
        "paw",
        "1",
        "2",
        "3",
      ]
    }
  />
  <form
    className="submit_game_command"
    onChange={[Function]}
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Submit Comand
  </button>
</div>
`;
